Dockerfile update to debian 13 "trixie"

thaJeztah opened the pull request:
Update our Dockerfile to use debian 13 as base. Some notes about architectures from the [announcement][1];

> This release for the first time officially supports the riscv64 architecture,
> allowing users to run Debian on 64-bit RISC-V hardware and benefit from all
> Debian 13 features. A total of seven architectures are officially supported
> for trixie:
>
> - 64-bit PC (amd64),
> - 64-bit ARM (arm64),
> - ARM EABI (armel),
> - ARMv7 (EABI hard-float ABI, armhf),
> - 64-bit little-endian PowerPC (ppc64el),
> - 64-bit little-endian RISC-V (riscv64),
> - IBM System z (s390x)
>
> i386 is no longer supported as a regular architecture: there is no official
> kernel and no Debian installer for i386 systems. The i386 architecture is
> now only intended to be used on a 64-bit (amd64) CPU. Users running i386
> systems should not upgrade to trixie. Instead, Debian recommends either
> reinstalling them as amd64, where possible, or retiring the hardware.
>
> trixie will be the last release for the armel architecture. See
> [5.1.3. Last release for armel][2] in the release notes for more information
> on our ARM EABI support.

[1]: https://www.debian.org/News/2025/20250809
[2]: https://www.debian.org/releases/trixie/release-notes/issues.html#last-release-for-armel

<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/moby/moby/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**

**- How I did it**

**- How to verify it**

**- Human readable description for the release notes**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog.
It must be placed inside the below triple backticks section.

NOTE: Only fill this section if changes introduced in this PR are user-facing.
The PR must have a relevant impact/ label.
-->
```markdown changelog

```

**- A picture of a cute animal (not mandatory but encouraged)**

thaJeztah committed:
Dockerfile update to debian 13 "trixie"

Update our Dockerfile to use debian 13 as base. Some notes about architectures
from the [announcement][1];

> This release for the first time officially supports the riscv64 architecture,
> allowing users to run Debian on 64-bit RISC-V hardware and benefit from all
> Debian 13 features. A total of seven architectures are officially supported
> for trixie:
>
> - 64-bit PC (amd64),
> - 64-bit ARM (arm64),
> - ARM EABI (armel),
> - ARMv7 (EABI hard-float ABI, armhf),
> - 64-bit little-endian PowerPC (ppc64el),
> - 64-bit little-endian RISC-V (riscv64),
> - IBM System z (s390x)
>
> i386 is no longer supported as a regular architecture: there is no official
> kernel and no Debian installer for i386 systems. The i386 architecture is
> now only intended to be used on a 64-bit (amd64) CPU. Users running i386
> systems should not upgrade to trixie. Instead, Debian recommends either
> reinstalling them as amd64, where possible, or retiring the hardware.
>
> trixie will be the last release for the armel architecture. See
> [5.1.3. Last release for armel][2] in the release notes for more information
> on our ARM EABI support.

[1]: https://www.debian.org/News/2025/20250809
[2]: https://www.debian.org/releases/trixie/release-notes/issues.html#last-release-for-armel

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah commented:
Also need to update the frozen images, but letting this one run as-is first to see if anything explodes.

We don't need to update yet, but I wanted at least to have a (draft) PR to see if everything works out fine for our CI.

thaJeztah committed:
Dockerfile: install e2fsprogs to fix missing mkfs.ext4

It's used in at least one test;

    === FAIL: amd64.integration-cli TestDockerDaemonSuite/TestDaemonNoSpaceLeftOnDeviceError (0.11s)
        docker_cli_daemon_test.go:1065: assertion failed:
            Command:  mkfs.ext4 -F /tmp/no-space-left-on-device-test183525913/testfs.img
            ExitCode: 127
            Error:    exec: "mkfs.ext4": executable file not found in $PATH
            Stdout:
            Stderr:

            Failures:
            ExitCode was 127 expected 0
            Expected no error
        --- FAIL: TestDockerDaemonSuite/TestDaemonNoSpaceLeftOnDeviceError (0.11s)

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

akerouanton commented:
I cherry-picked the commit from https://github.com/moby/moby/pull/50745 to double-check that iptablesdoc tests are passing here. I'll take a look at the nftabler failures.

akerouanton committed:
nftablesdoc: stringify numerical dstnat prio  (bis)

Signed-off-by: Albin Kerouanton <albinker@gmail.com>
Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

akerouanton committed:
d/libnet: TestUserChain: fix error matching for nonexistent chains

Debian 13 ships iptables-nft v1.8.11 which returns a different error
when `iptables -S` a nonexistent chain than previous versions.

Older versions:

  ip6tables v1.8.9 (nf_tables): chain `<chain>' in table `filter' is incompatible, use 'nft' tool.

Newer versions:

  ip6tables: No chain/target/match by that name.

Signed-off-by: Albin Kerouanton <albinker@gmail.com>
Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

akerouanton committed:
fixup! d/libnet: TestUserChain: fix error matching for nonexistent chains

Signed-off-by: Albin Kerouanton <albinker@gmail.com>
Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah commented:
Hm, ok, looks like we need to update the regex, or perhaps some simpler parsing (splitting the fields, then find the version string);


```
=== Failed
=== FAIL: daemon/libnetwork TestUserChain (0.00s)
    firewall_linux_test.go:155: assertion failed: len(match) is 4: could not determine iptables version from "iptables v1.8.11 (nf_tables)\n"
```

akerouanton committed:
fixup! fixup! d/libnet: TestUserChain: fix error matching for nonexistent chains

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

ctalledo commented:
Rebased onto current master and opened as #53354 (from @ctalledo's fork, since push access to this fork isn't available). All test failures that were blocking this PR had already landed in master via separate PRs — no extra changes from master were needed:

| Fix | Landed via |
|-----|-----------|
| `TestBridgeICC` / `TestBridgeINC` firewalld race | #50819 |
| `TestUserChain` iptables version parsing | #50841 |
| nftablesdoc `dstnat` priority format | #50745 |
| `extra_status.go` unallocated network regression | #53325 |

Also fixed a bug introduced by one of the fixup commits: `versionLt` had been changed to `parsedMajor <= major && parsedMinor <= minor && parsedPatch < patch`, which gives wrong results when `minor < threshold_minor` but `patch > threshold_patch` (e.g. v1.7.15 incorrectly reported as >= v1.8.10).

Commits were squashed to 3 logical units in #53354.
